### DISCUSSION TITLE
Able to pass runtime argument to task decorator?

### DISCUSSION THREAD

**Discussion by aamster** [apache/airflow#29046] [Q&A]
I am trying to decorate a task_sensor at runtime by passing in a custom timeout

```
import datetime

from airflow.decorators import task, task_group
from airflow.models import Param
from airflow.models.dag import dag
from airflow.sensors.base import PokeReturnValue


def f(timeout):
    @task.sensor(timeout=timeout)
    def inner():
        print('timeout', timeout)
        return PokeReturnValue(is_done=True)
    return inner

@dag(
    dag_id="foo",
    schedule=None,
    start_date=datetime.datetime.now(),
    params={'timeout': Param(default=None)}
)
def foo():
    @task_group
    def g():
        f(timeout='{{ params.timeout }}')()

    g()


foo()


if __name__ == '__main__':
    foo().test(run_conf={'timeout': 1})


```

This gives
```
airflow.exceptions.AirflowException: The timeout must be a non-negative number

```
because the string `{{ params.timeout }}` is being passed instead.
It doesnt seem possible to pass a runtime argument to a regular function in order to pass it to the decorator. Is it possible some other way?

**Comment by Taragolis** [comment]
It is not possible in current implementation of BaseSensorOperator.

There is no [templated_fields](https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html#jinja-templating) defined in BaseSensorOperator, all evaluation from templates/XcomArgs to actual values happen after `__init__` and before `execute`, however validation happen in class constructor `__init__`
